src/frontend/src/functions/forms.tsx

```diff
@@ -96,7 +191,28 @@ export function openModalApiForm(props: ApiFormProps) {
     return;
   }
 
-  let url = constructFormUrl(props);
+  // Generate a random modal ID for controller
+  let modalId: string =
+    `modal-${props.title}-${props.url}-${props.method}` + generateUniqueId();
+
+  props.actions = [
+    ...(props.actions || []),
+    {
+      text: props.cancelText ?? t`Cancel`,
+      color: props.cancelColor ?? 'blue',
+      onClick: () => {
+        modals.close(modalId);
+      }
+    }
+  ];
+
+  const oldFormSuccess = props.onFormSuccess;
```

Comment:
Why is this called "old" here - is this intentional?

Reply:
I answered this on the other PR in this [comment](https://github.com/inventree/InvenTree/pull/5897#discussion_r1399180113)